pygmt/tests/test_clib_virtualfiles.py

```diff
@@ -0,0 +1,384 @@
+"""
+Test the API functions related to virtual files.
+"""
+from contextlib import contextmanager
+from itertools import product
+
+import numpy as np
+import pandas as pd
+import pytest
+import xarray as xr
+from pygmt import clib
+from pygmt.exceptions import GMTCLibError, GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+
+
+@contextmanager
+def mock(session, func, returns=None, mock_func=None):
```

Comment:
Please note that the same code is also defined in `test_clib.py`.

Reply:
Hmm, the typical way of sharing *fixtures* in pytest would be to put them in [`conftest.py`](https://docs.pytest.org/en/7.1.x/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files), but not sure about this contextmanager... We could also just keep `test_virtual_file_fails` in `test_clib.py` to avoid duplicating this contextmanager?

Reply:
I just realized the `mock` function is also used in other files like `test_clib_put_matrix.py` by `from pygmt.tests.test_clib import mock`. 

This may not be a best practice but it works. Done in d5ef26eb.